docs/manual/src/Upgrading.md

```diff
@@ -0,0 +1,37 @@
+# v0.28.x -> v0.29.x
+
+## Custom types
+
+Custom types are now implemented using a macro, rather than implementing the
+`UniffiCustomTypeConverter` trait.
+
+Before:
+
+```rust
+impl UniffiCustomTypeConverter for MyType {
+    type Builtin = UniFFIBuiltinType;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        ...
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        ...
+    }
+}
+```
+
+After:
+
+```
+uniffi::custom_type!(MyType, UniFFIBuiltinType, {
+    try_from_existing: |val| { Ok(...) },
+    into_existing: |obj| { ... },
+})
+```
+
+The custom_type macro is more flexible than the old system.  For example, the `try_from_existing`
+and `into_existing` can be omitted in many cases.  If `try_from_existing` is omitted, then the
+`TryFrom<UniFFIBuiltinType>` will be used instead.  Likewise, if `into_existing` is omitted, then
+`From<MyType>` will be used instead. See the [Custom Types](./udl/custom_types.md) for details and
+other features of the new macro.
```

Comment:
Thanks for updating these docs. As above though, if we stick with the term "existing" we should try and remove `UniFFIBuiltinType` entirely from these docs. Maybe even replace `MyType` with `NewCustomType` or something to fully reinforce this? I think that's largely independent of how we name the 2 methods - IIUC, we want to avoid the term "builtin" regardless, right?

So the very first part of the docs is perfect - ie, "If `try_from_existing` is omitted, then `TryFrom<AnExistingUniFFIType>` will be used" seems completely logical.

However, the second part is a little less so. "Likewise, if `into_existing` is omitted, then
`From<NewCustomType>` will be used instead." doesn't have quite the same symmetry. Should this then be `from_custom`?

That would leave us with:
"""
uniffi::custom_type!(NewCustomType, AnExistingUniFFIType, {
    try_from_existing: |val| { Ok(...) },
    from_custom: |obj| { ... },
})

The custom_type macro is more flexible than the old system. For example, `try_from_existing` and `from_custom` can be omitted in many cases. If `try_from_existing` is omitted, then `TryFrom<AnExistingUniFFIType>` will be used instead. Likewise, if `from_custom` is omitted, then `From<NewCustomType>` will be used instead.
"""

which seems to better capture reality and makes it much clearer about what the user is expected to implement to make this work, even if it is missing the symmetry of from/info.

WDYT?

Reply:
Nitpick: Should consistently use `UniFfi` in Rust type namespace.

Reply:
One more option that I like is using the names `try_from_existing` and `from_custom`, since Rust encourages users to implement `From` and also using the less-specific description from the 3rd option.